docs/_packages/oauth.md

```diff
@@ -62,17 +62,47 @@ const installer = new InstallProvider({
+You'll need an installation URL when you want to test your own installation in order to submit your app to the App Directory and in case you need additional authorizations (such as user tokens) from users inside a team where your app is already installed. These URLs are also commonly used on your own webpages as the link for an ["Add to Slack" button](https://api.slack.com/docs/slack-button).
 
-The `installProvider.generateInstallUrl()` method will create an installation URL for you. It takes in an options argument which at a minimum contains a `scopes` property. `installProvider.generateInstallUrl()` options argument also supports `metadata`, `teamId`, `redirectUri` and `userScopes` properties.
+There are two approaches the `InstallProvider` provides to show an installation page to the user prior to beginning the OAuth process:
+
+1. You can let `InstallProvider` render the installation page at a URL/path of your choosing [using the `handleInstallPath()` method](#using-handleinstallpath). It will automatically display an "Add to Slack" button and encode any desired user or bot scopes and metadata you specify.
+2. You can render the installation page yourself and [call `generateInstallUrl()`](#using-generateinstallurl) to generate a secure slack.com-based URL to initiate the OAuth process. You would use this generated URL in your own installation page's links/buttons/UI elements.
+
+#### Using `handleInstallPath`
+
+If you don't need to customize the installation page users will be shown, you can let this package render the installation page for you using the `handleInstallPath()` method.
 
 ```javascript
-installer.generateInstallUrl({
-  // Add the scopes your app needs
-  scopes: ['channels:read']
-})
+// Assume the installation page is located at /slack/install
+app.get('/slack/install', async (req, res) => {
+  await installer.handleInstallPath(req, res, {
+    scopes: ['chat:write'],
+    userScopes: ['channels:read'],
+    metadata: 'some_metadata',
+  });
+});
+```
+
+The `handleInstallPath` method accepts an options object as its third argument which supports `scopes`, `metadata`, `userScopes`, `teamId` and `redirectUri` properties (check out the [source code for this interface](https://github.com/slackapi/node-slack-sdk/blob/main/packages/oauth/src/install-url-options.ts) for more details).
+
+#### Using `generateInstallUrl`
+
+If you want to customize the installation page users will be shown, you may generate an installation URL dynamically and use the generated URL as part of the installation page displayed to the user.
```

Comment:
As I mentioned above, we don't need to mention `generateInstallUrl` for this reason. We can suggest `renderHtmlForInstallPath` instead. We still can have `generateInstallUrl` section for backward-compatibility and advanced use cases to take 100% control of the endpoint handling.

Reply:
@seratch I've now changed this section to singularly focus on `handleInstallPath`, and moved any references to the old `generateInstallUrl` way under the details section. Hopefully that encourages developers down our recommended path (resolving this for now).